scripts/benchmark_db/bench.rb

```diff
@@ -176,169 +211,434 @@ def snapshot_height(snapshot)
+    puts '(I) Downloading snapshot...'
+    syscall('aria2c', '-x5', url, chdir: dir)
+    puts "(I) Decompressing #{filename}..."
+    syscall('zstd', '-d', filename, chdir: dir)
+    puts '(I) Verifying...'
+    syscall('sha256sum', '--check', '--status', checksum_filename, chdir: dir)
+
+    FileUtils.mv("#{dir}/#{decompressed_filename}", output_dir)
+  end
+  "#{output_dir}/#{decompressed_filename}"
+end
+
+def download_snapshot(output_dir: WORKING_DIR, chain: 'calibnet', url: nil)
+  puts "output_dir: #{output_dir}"
+  puts "chain: #{chain}"
+  url = get_url(chain, url)
+  puts "snapshot_url: #{url}"
+
+  download_and_move(url, output_dir)
+end
+
+# Base benchmark class
 class Benchmark
   attr_reader :name, :metrics
-  attr_accessor :snapshot_path, :heights
-
-  def db_size
-    config_path = "#{TEMP_DIR}/#{@name}.toml"
+  attr_accessor :dry_run, :snapshot_path, :heights, :chain
 
-    line = syscall('./target/release/forest-cli', '-c', config_path, 'db', 'stats').split("\n")[1]
-    match = line.match(/Database size: (.+)/)
-    match[1]
+  def initialize(name:, config: {})
+    @name = name
+    @config = config
   end
 
-  def clean_db(dry_run)
-    puts 'Wiping db'
+  def exec_command(command, benchmark = nil)
+    puts "$ #{command.join(' ')}"
+    return {} if @dry_run
 
-    config_path = "#{TEMP_DIR}/#{@name}.toml"
-
-    syscall('./target/release/forest-cli', '-c', config_path, 'db', 'clean', '--force') unless dry_run
+    metrics = Concurrent::Hash.new
+    t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+    exec_command_aux(command, metrics, benchmark)
+    t1 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+    metrics[:elapsed] = hr(t1 - t0)
+    metrics
   end
 
   def build_config_file
     config = @config.deep_merge(default_config)
-    config_path = "#{TEMP_DIR}/#{@name}.toml"
+    config_path = "#{data_dir}/#{@name}.toml"
 
     toml_str = TomlRB.dump(config)
-    File.open(config_path, 'w') { |file| file.write(toml_str) }
+    File.write(config_path, toml_str)
   end
   private :build_config_file
 
-  def build_substitution_hash(dry_run)
-    snapshot = @snapshot_path
-    height = snapshot_height(snapshot)
+  def build_substitution_hash
+    height = snapshot_height(@snapshot_path)
     start = height - @heights
 
-    return { c: '<tbd>', s: '<tbd>', h: start } if dry_run
-
-    config_path = "#{TEMP_DIR}/#{@name}.toml"
+    return { c: '<tbd>', s: '<tbd>', h: start } if @dry_run
 
-    snapshot_path = File.file?(snapshot) ? snapshot : "#{snapshot_dir}/#{snapshot}"
+    config_path = "#{data_dir}/#{@name}.toml"
 
-    { c: config_path, s: snapshot_path, h: start }
+    { c: config_path, s: @snapshot_path, h: start }
   end
   private :build_substitution_hash
 
-  def build_artefacts(dry_run)
-    # exec_command(%w[cargo clean], dry_run)
-    exec_command(build_command, dry_run)
+  def build_client
+    if Dir.exist?(repository_name)
+      puts "(W) Directory #{repository_name} is already present"
```

Comment:
If I understand correctly, `W` and `I` are kind of log levels? If so, how about we add a proper logger like in sync check https://github.com/ChainSafe/forest-iac/blob/main/scripts/sync_check/sync_check_process.rb#L27?